### ISSUE TITLE
Support HTTPS in etcd URL schemes

### ISSUE THREAD

**Issue by loshkovskyi** [projectcalico/calico-docker#512]
Right now it's not possible to use Calico with etcd cluster where all URLs use HTTPS. It would be great to have such option for security implications. The same is for Calico Kubernetes plugin, as everything I can do is to change the `ETCD_AUTHORITY` variable which is not enough to access the etcd store.

**Comment by lxpollitt** [comment]
+1

**Comment by tomdee** [comment]
We should prioritise fixing this. The code should be easy but we need to agree on a design. Any input would be appreciated.

One option would be to use another environment variable such as `ETCD_SCHEME` which defaults to `http` but can also take the value `https`

@loshkovskyi What are your thoughts?

**Comment by loshkovskyi** [comment]
Here is an example how I can access etcd store over HTTPS:
```
etcdctl --cert-file /etc/origin/master/master.etcd-client.crt \
--key-file /etc/origin/master/master.etcd-client.key \
--ca-file /etc/origin/master/master.etcd-ca.crt \
--peers https://osv3-master-01.example.com:2379,https://osv3-node-01.example.com:2379,https://osv3-node-02.example.com:2379 \
ls
```
So if I set `ETCD_AUTHORITY` to `https`, I need several more variables to set paths to the certificates.

**Comment by robbrockbank** [comment]
I've been looking into a design for this.  Please touch base with me before tackling this one.

**Comment by TrimBiggs** [comment]
Fixed by #654
